### PR TITLE
ExpansionTile audit

### LINKED ISSUE #74963: ExpansionTile accessibility information isn't read out in VoiceOver (iOS)
It seems that https://github.com/flutter/flutter/pull/20766 only fixed the issue for ExpansionPanel - when I use voiceover on an ExpansionTile widget (on iOS), it only reads out the pre-expanded content (and doesn't give any hints to the user that the tile can be expanded/collapsed).

I can link the YAQS/Buganizer if that will help?

B

### LINKED ISSUE #120650: When interacting with a ExpansionTile using a screen reader, the status of the tile is not announced
Internal bug: b/269139224

(Copied from the internal bug above)

The bug also happens on iOS devices.

## Steps to Reproduce

1. With an Android device or emulator with Talkback enabled.
2. Focus on the collapsed/expanded ExpansionTile and verify what the screen reader announces.
3. Double click to activate/collapse the ExpansionTile and verify the screen reader announcement.

**Expected results:** 

The tile should read "double tap to expand", "double tap to collapse" when focused on it.
The tile should announce "expanded" or "collapsed" when clicking on it.

**Actual results:** 

Screen reader is just reading "double tap to activate" when focused on the tile.
The tile doesn't announce anything when it's expanded or collapsed.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        // We also apply a custom font, ProductSans.
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: ExpansionTile(
          title: Text('Tile'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

### LINKED ISSUE #121636: [accessibility] VoiceOver on toggleable Material widgets on iOS
Internal: b/275699766

While fixing https://github.com/flutter/flutter/issues/120650 and investigating https://github.com/flutter/flutter/issues/74963, I noticed that regular `ListTile` doesn't announce any action and other toggleable material widgets with voice-over doesn't sound right compared to Android. 

Since these toggleable widgets are integrable parts of list tile-dependent widgets such as `CheckboxListTile`, `SwitchListTile`, `RadioListTile`, and `ExpansionTile`. I wanna make sure the hints are consistent on iOS for all these widgets.

Here I created an example that showcases toggleable Material widgets and their list tile variants.

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(useMaterial3: true),
      home: const Example(),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  bool toggle = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: SingleChildScrollView(
          child: Column(
            children: <Widget>[
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  Column(
                    children: <Widget>[
                      const DescriptionText(text: 'Switch'),
                      Switch(
                        value: toggle,
                        onChanged: (bool value) {
                          setState(() {
                            toggle = !toggle;
                          });
                        },
                      ),
                    ],
                  ),
                  Column(
                    children: <Widget>[
                      const DescriptionText(text: 'Checkbox'),
                      Checkbox(
                        value: toggle,
                        onChanged: (bool? value) {
                          setState(() {
                            toggle = !toggle;
                          });
                        },
                      ),
                    ],
                  ),
                  Column(
                    children: <Widget>[
                      const DescriptionText(text: 'Radio'),
                      Radio<int>(
                        value: toggle ? 1 : 2,
                        groupValue: 1,
                        toggleable: true,
                        onChanged: (int? value) {
                          setState(() {
                            toggle = !toggle;
                          });
                        },
                      ),
                    ],
                  )
                ],
              ),
              const Divider(),
              const DescriptionText(text: 'ListTile'),
              ListTile(
                title: const Text('Headline'),
                trailing: const Icon(Icons.favorite),
                selected: toggle,
                onTap: () {
                  setState(() {
                    toggle = !toggle;
                  });
                },
              ),
              const Divider(),
              const DescriptionText(text: 'SwitchListTile'),
              SwitchListTile(
                title: const Text('Headline'),
                value: toggle,
                onChanged: (bool value) {
                  setState(() {
                    toggle = !toggle;
                  });
                },
              ),
              const Divider(),
              const DescriptionText(text: 'CheckboxListTile'),
              CheckboxListTile(
                title: const Text('Headline'),
                value: toggle,
                onChanged: (bool? value) {
                  setState(() {
                    toggle = !toggle;
                  });
                },
              ),
              const Divider(),
              const DescriptionText(text: 'RadioListTile'),
              RadioListTile<int>(
                title: const Text('Headline'),
                value: toggle ? 1 : 2,
                groupValue: 1,
                toggleable: true,
                onChanged: (int? value) {
                  setState(() {
                    toggle = !toggle;
                  });
                },
              ),
              const Divider(),
              const DescriptionText(text: 'ExpansionTile'),
              const ExpansionTile(
                title: Text('Headline'),
                children: <Widget>[FlutterLogo(size: 50)],
              ),
              const Divider(),
            ],
          ),
        ),
      ),
    );
  }
}

class DescriptionText extends StatelessWidget {
  const DescriptionText({super.key, required this.text});

  final String text;

  @override
  Widget build(BuildContext context) {
    return ExcludeSemantics(child: Text(text));
  }
}

``` 
	
</details>

### Preview
https://user-images.githubusercontent.com/48603081/221981458-bc13a451-a351-4c08-a859-46b3327071cc.MP4

https://user-images.githubusercontent.com/48603081/221981497-6cc2ee05-d4c7-4d05-ac78-3d9128333b68.MP4

### Observations

- All toggleable widgets are announced as 'switch button" on iOS.
- No double tap to expand announcement for expansion tile https://github.com/flutter/flutter/issues/74963
- The radio button is never announced due to https://github.com/flutter/flutter/issues/80182

Fix voiceover issues for the following widgets:

- [ ] [Switch](https://api.flutter.dev/flutter/material/Switch-class.html)
- [ ] [Checkbox](https://api.flutter.dev/flutter/material/Checkbox-class.html)
- [ ] [Radio](https://api.flutter.dev/flutter/material/Radio-class.html)
- [ ] [CheckboxListTile](https://api.flutter.dev/flutter/material/CheckboxListTile-class.html)
- [ ] [SwitchListTile](https://api.flutter.dev/flutter/material/SwitchListTile-class.html)
- [ ] [RadioListTile](https://api.flutter.dev/flutter/material/RadioListTile-class.html)
- [ ] [ExpansionTile](https://api.flutter.dev/flutter/material/ExpansionTile-class.html)

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/121636
fixes https://github.com/flutter/flutter/issues/120650
fixes https://github.com/flutter/flutter/issues/74963

This is basically the same pr as https://github.com/flutter/flutter/pull/121699

The radio button fix will be in the engine

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by TahaTesser — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/material/material_localizations.dart [hunk 1]
```diff
@@ -452,6 +452,24 @@ abstract class MaterialLocalizations {
   /// The semantics hint to describe the tap action on a collapsed [ExpandIcon].
   String get collapsedIconTapHint => 'Expand';
 
+  /// The semantics hint to describe the tap action on an expanded [ExpansionTile] on iOS and macOS. This is appended to the [collapsedHint] hint to provide a more detailed description of the action, e.g. "Expanded double tap to collapse".
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/material_localizations.dart:455]: Wrap the comment at 80 chars.

File: packages/flutter/lib/src/material/material_localizations.dart [hunk 2]
```diff
@@ -452,6 +452,24 @@ abstract class MaterialLocalizations {
...
+  String get expansionTileExpandedHint => 'double tap to collapse';
+
+  /// The semantics hint to describe the tap action on a collapsed [ExpansionTile] on iOS and macOS. This is appended to the [expandedHint] hint to provide a more detailed description of the action, e.g. "Collapsed double tap to expand".
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/material_localizations.dart:458]: Wrap the comment at 80 chars.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**